discovery_engine_core/ai/bert/src/pipeline.rs

```diff
@@ -33,7 +35,7 @@ use crate::{
 pub struct Pipeline<K, P> {
     pub(crate) tokenizer: Tokenizer,
     pub(crate) model: Model<K>,
-    pub(crate) pooler: P,
+    pooler: PhantomData<P>,
```

Comment:
we could further simplify this, eg with a `Pipeline` trait and a `Pipeline::Pooler` associated type, but this would be out of scope of this pr.